server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java

```diff
@@ -303,6 +304,12 @@ private GenericContainer<?> createContainer(int i, Consumer<OutputFrame>... logC
                   cmd.getHostConfig().withMemorySwap(IMAGE_MEMORY_SWAP);
                }
             });
+      if (OS.getCurrentOs().equals(OS.MAC_OS) || OS.getCurrentOs().equals(OS.WINDOWS)) {
```

Comment:
What happens if we start multiple containers ?

Reply:
this is supposed to work with the JUnit Extension to run a single container and help users have a listening infinispan server running for testing.
If we need multiple containers, or a cluster, this means we are in another level of needs on testing

Reply:
the problem I see since yesterday is that we mix our need to created units tests and test infinispan behaviour with users that are infinispan users and might need to validate their functional code that works with infinispan in a easy way.

Reply:
I understand you. But `ContainerInfinispanServerDriver` was designed for our internal needs and was repurposed for what you described. Maybe add a condition to that `if` statement to also ensure it only applies when `configuration.numServers() == 1`